cmd/query/app/grpc_handler_test.go

```diff
@@ -0,0 +1,481 @@
+			},
+			{
+				TraceID: mockTraceID,
+				SpanID:  model.NewSpanID(7),
+				Process: &model.Process{},
+			},
+			{
+				TraceID: mockTraceID,
+				SpanID:  model.NewSpanID(8),
+				Process: &model.Process{},
+			},
+			{
+				TraceID: mockTraceID,
+				SpanID:  model.NewSpanID(9),
+				Process: &model.Process{},
+			},
+			{
+				TraceID: mockTraceID,
+				SpanID:  model.NewSpanID(10),
+				Process: &model.Process{},
+			},
+			{
+				TraceID: mockTraceID,
+				SpanID:  model.NewSpanID(11),
+				Process: &model.Process{},
+			},
+		},
+		Warnings: []string{},
+	}
+)
+
+func newGRPCServer(t *testing.T, q *querysvc.QueryService, logger *zap.Logger, tracer opentracing.Tracer) (*grpc.Server, net.Addr) {
+	lis, _ := net.Listen("tcp", grpcServerPort)
+	grpcServer := grpc.NewServer()
+	grpcHandler := NewGRPCHandler(*q, logger, tracer)
+	api_v2.RegisterQueryServiceServer(grpcServer, grpcHandler)
+
+	go func() {
+		err := grpcServer.Serve(lis)
+		require.NoError(t, err)
+	}()
+
+	return grpcServer, lis.Addr()
+}
+
+func newGRPCClient(t *testing.T, addr net.Addr) (api_v2.QueryServiceClient, *grpc.ClientConn) {
+	conn, err := grpc.Dial(addr.String(), grpc.WithInsecure())
+	require.NoError(t, err)
+	return api_v2.NewQueryServiceClient(conn), conn
+}
+
+func newQueryService(options querysvc.QueryServiceOptions) (*querysvc.QueryService, *spanstoremocks.Reader, *depsmocks.Reader) {
+	readStorage := &spanstoremocks.Reader{}
+	dependencyStorage := &depsmocks.Reader{}
+	return querysvc.NewQueryService(readStorage, dependencyStorage, options), readStorage, dependencyStorage
+}
+
+func initializeTestServerGRPC(t *testing.T) (*grpc.Server, net.Addr, *spanstoremocks.Reader, *depsmocks.Reader) {
+	q, readStorage, dependencyStorage := newQueryService(querysvc.QueryServiceOptions{})
+	logger := zap.NewNop()
+	tracer := opentracing.NoopTracer{}
+
+	grpcServer, addr := newGRPCServer(t, q, logger, tracer)
+
+	return grpcServer, addr, readStorage, dependencyStorage
+}
+
+func initializeTestServerGRPCWithOptions(t *testing.T) (*grpc.Server, net.Addr, *spanstoremocks.Reader, *depsmocks.Reader, *spanstoremocks.Reader, *spanstoremocks.Writer) {
+	archiveSpanReader := &spanstoremocks.Reader{}
+	archiveSpanWriter := &spanstoremocks.Writer{}
+	q, readStorage, dependencyStorage := newQueryService(querysvc.QueryServiceOptions{
+		ArchiveSpanReader: archiveSpanReader,
+		ArchiveSpanWriter: archiveSpanWriter,
+	})
+	logger := zap.NewNop()
+	tracer := opentracing.NoopTracer{}
+
+	grpcServer, addr := newGRPCServer(t, q, logger, tracer)
+
+	return grpcServer, addr, readStorage, dependencyStorage, archiveSpanReader, archiveSpanWriter
+}
+
+func TestGetTraceSuccessGRPC(t *testing.T) {
+	server, addr, readMock, _ := initializeTestServerGRPC(t)
+	defer server.Stop()
+	readMock.On("GetTrace", mock.AnythingOfType("*context.valueCtx"), mock.AnythingOfType("model.TraceID")).
+		Return(mockTrace, nil).Once()
+
+	client, conn := newGRPCClient(t, addr)
+	defer conn.Close()
```

Comment:
similar to above, `withServerAndClient(t *testing.T, func actualTest(server *server, client *client))`

this way it would remove a lot of boilerplate code from the tests